<a href="https://colab.research.google.com/github/qitinnstud/myfirstexperience/blob/For_homework/For_homework1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Импорт библиотеки pandas

In [ ]:
import pandas as pd
#from google.colab import files
#uploaded = files.upload()
#import io
#df = pd.read_csv(io.BytesIO(uploaded['Weather Data.csv']))

Монтирую google drive

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Читаю DataFrame в переменную df и вывожу первые 11 строк

In [ ]:
df=pd.read_csv("drive/MyDrive/Weather_Data1.csv", sep='#');
df.head()

Date/Time Temp_C Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0  1/1/2012 0:00   -1,8             -3,9         86                4   
1  1/1/2012 1:00   -1,8             -3,7         87                4   
2  1/1/2012 2:00   -1,8             -3,4         89                7   
3  1/1/2012 3:00   -1,5             -3,2         88                6   
4  1/1/2012 4:00   -1,5             -3,3         88                7   

  Visibility_km Press_kPa               Weather  
0           8,0    101,24                   Fog  
1           8,0    101,24                   Fog  
2           4,0    101,26  Freezing Drizzle,Fog  
3           4,0    101,27  Freezing Drizzle,Fog  
4           4,8    101,23                   Fog

In [ ]:
df.to_csv("drive/MyDrive/Weather_Data1.csv",sep='#',index=False)

Запрос инфо датафрейма. Вывод информации о колонках (заголовках) и типа данных в них:

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8784 entries, 0 to 8783
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Date/Time         8784 non-null   object
 1   Temp_C            8784 non-null   object
 2   Dew Point Temp_C  8784 non-null   object
 3   Rel Hum_%         8784 non-null   int64 
 4   Wind Speed_km/h   8784 non-null   int64 
 5   Visibility_km     8784 non-null   object
 6   Press_kPa         8784 non-null   object
 7   Weather           8784 non-null   object
dtypes: int64(2), object(6)
memory usage: 549.1+ KB


Запрос статистики датафрейма (число строк, среднее, СТО, мин, макс. и квартили):

In [ ]:
df.describe()

Temp_C  Dew Point Temp_C    Rel Hum_%  Wind Speed_km/h  \
count  8784.000000       8784.000000  8784.000000      8784.000000   
mean      8.798144          2.555294    67.431694        14.945469   
std      11.687883         10.883072    16.918881         8.688696   
min     -23.300000        -28.500000    18.000000         0.000000   
25%       0.100000         -5.900000    56.000000         9.000000   
50%       9.300000          3.300000    68.000000        13.000000   
75%      18.800000         11.800000    81.000000        20.000000   
max      33.000000         24.400000   100.000000        83.000000   

       Visibility_km    Press_kPa  
count    8784.000000  8784.000000  
mean       27.664447   101.051623  
std        12.622688     0.844005  
min         0.200000    97.520000  
25%        24.100000   100.560000  
50%        25.000000   101.070000  
75%        25.000000   101.590000  
max        48.300000   103.650000

включая стр. тип object:

In [ ]:
df.describe(include='object')

Date/Time       Weather
count            8784          8784
unique           8784            50
top     1/1/2012 0:00  Mainly Clear
freq                1          2106

Дата-время представлиены типом 'object'. "Преимущественно ясно" встречается чаще - 2106 вхождений.